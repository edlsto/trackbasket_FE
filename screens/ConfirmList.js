import React, { useContext } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { Button } from '../components/Button';
import VolunteerContext from '../volunteer-context';
import moment from 'moment';

const ConfirmList = ({ navigation, route }) => {
  const item = route.params;
  const { assignedLists, setAssignedLists } = useContext(VolunteerContext);

  const handleSubmit = () => {
    setAssignedLists([item, ...assignedLists]);
    navigation.navigate('VolunteerHome');
  };

  return (
    <View style={styles.container}>
      <View style={styles.containerInner}>
        <View style={styles.orderContainer}>
          <View style={styles.userDetails}>
            <Text style={styles.introtext}>Order details</Text>
            <Text style={styles.orderText}>
              <Text style={styles.orderTextBold}>Name: </Text>
              {route.params.userDetails.name}
            </Text>
            <Text style={styles.orderText}>
              <Text style={styles.orderTextBold}>Delivery address: </Text>
              {route.params.userDetails.address}
            </Text>
            <Text style={styles.orderText}>
              <Text style={styles.orderTextBold}>Store:</Text>{' '}
              {route.params.storeId}
            </Text>
            <Text style={styles.orderText}>
              <Text style={styles.orderTextBold}>Distance from you:</Text>{' '}
              {route.params.distance.toFixed(2)} miles
            </Text>
          </View>
          <View style={styles.orderInfo}>
            <Text style={styles.orderText}>
              <Text style={styles.orderTextBold}>Ordered:</Text>{' '}
              {moment(route.params.created_at).format('MMM D')}
            </Text>
            <Text style={styles.orderText}>
              <Text style={styles.orderTextBold}>Items:</Text>{' '}
              {route.params.number_items}
            </Text>
          </View>
        </View>
        <Text style={styles.introtext}>Do you want to select this list?</Text>
        <Button text="CONFIRM" onPress={handleSubmit} />
        <Button
          text="CANCEL"
          customStyles={{ backgroundColor: '#DE5959', marginTop: 10 }}
          onPress={() => navigation.navigate('SelectList')}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  introtext: {
    fontSize: 30,
    marginBottom: 20,
    textAlign: 'center',
  },
  containerInner: {
    justifyContent: 'center',
    width: 300,
    marginBottom: 100,
  },
  userDetails: {
    marginBottom: 10,
  },
  orderContainer: {
    marginBottom: 30,
  },
  orderText: {
    fontSize: 20,
  },
  orderTextBold: {
    fontWeight: 'bold',
  },
});

export default ConfirmList;
