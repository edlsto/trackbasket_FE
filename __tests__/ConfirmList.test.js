import React from 'react';
import ConfirmList from '../screens/ConfirmList';
import renderer from 'react-test-renderer';

test('renders correctly', async () => {
  const tree = renderer.create(<ConfirmList />).toJSON();
  expect(tree).toMatchSnapshot();
});
