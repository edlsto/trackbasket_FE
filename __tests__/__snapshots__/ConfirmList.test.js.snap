// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "center",
        "marginBottom": 100,
        "width": 300,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 30,
        }
      }
    >
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 30,
              "marginBottom": 20,
              "textAlign": "center",
            }
          }
        >
          Order details
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Name: 
          </Text>
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Delivery address: 
          </Text>
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Store:
          </Text>
           
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Distance from you:
          </Text>
           
           miles
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Ordered:
          </Text>
           
          Jun 1
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Items:
          </Text>
           
          ;
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#59DE7E",
          "opacity": 1,
          "paddingHorizontal": 40,
          "paddingVertical": 10,
          "width": 220,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "HelveticaNeue-Bold",
              "fontSize": 24,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Confirm
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#DE5959",
          "marginTop": 10,
          "opacity": 1,
          "paddingHorizontal": 40,
          "paddingVertical": 10,
          "width": 220,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "HelveticaNeue-Bold",
              "fontSize": 24,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Cancel
      </Text>
    </View>
  </View>
</View>
`;
